jwplatform/client.py

```diff
@@ -93,15 +100,29 @@ def request(self, method, path, body=None, headers=None, query_params=None):
 
         return self.raw_request(method=method, url=path, body=body, headers=headers)
 
+    def request_with_retry(self, method, path, body=None, headers=None, query_params=None,
```

Comment:
`http_connection_retry_count` could probably just be `retry_count` or `num_attempts` or something else here. Seems a bit long currently.

Also this method should probably have a docstring (it could mostly just be a copy of the one for request).